validphys2/src/validphys/n3fit_data.py

```diff
@@ -359,10 +359,20 @@ def pseudodata_table(groups_replicas_indexed_make_replica, replicas):
     `fitting::savepseudodata` is `true` (as per the default setting) and
     replicas are fitted one at a time. The table can be found in the replica
     folder i.e. <fit dir>/nnfit/replica_*/
-
     """
-    # Concatenate over replicas
-    df = pd.concat(groups_replicas_indexed_make_replica)
+    # groups_replicas_indexed_make_replica is collected over both replicas and dataset_input groups
+    # to correctly put this into a single dataframe, we first need to know the number of
+    # dataset_input groups there are for each replica
+    groups_per_replica = int(len(groups_replicas_indexed_make_replica) / len(replicas))
+    # then we make a list of pandas dataframes, each containing the pseudodata of all datasets
+    # generated for a singel replica
+    df = [
+        pd.concat(groups_replicas_indexed_make_replica[i : i + groups_per_replica])
+        for i in range(0, len(groups_replicas_indexed_make_replica), groups_per_replica)
+    ]
```

Comment:
Why can't you achieve this with a reshape or permutation of `groups_replicas_indexed_make_replica`

Reply:
because `groups_replicas_indexed_make_replica` is a list of `indexed_make_replica` containing a number of dataframes equal to number_of_replicas x number_of_datagroups. It's not a really clean input

Reply:
Just complicating your life here, but would it be possible to do something along the lines of 

`np.array(groups_replicas_indexed_make_replica).reshape(replicas, groups_per_replica)` ?

(or the other way around)

Reply:
No because it's a list of dataframes and I do need to retain the information on the labels